frontend/src/screens/apps/NewApp.tsx

```diff
@@ -57,25 +57,31 @@ const NewApp = () => {
     parseInt(maxAmountParam || "100000")
   );
 
+  // returns ISO string
   const parseExpiresParam = (expiresParam: string): string => {
     if (/\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}.\d{3}Z/.test(expiresParam)) {
       const d = new Date(expiresParam);
       const isIso =
         d instanceof Date &&
         !isNaN(d.getTime()) &&
-        d.toISOString() === expiresAtParam;
+        d.toISOString() === expiresParam;
```

Comment:
This `parseExpiresParam` method is way too complicated, why is it doing all these things?

Reply:
Can't the whole thing be replaced with:
```
return expiresParam ? new Date(parseInt(expiresParam) * 1000) : undefined;
```

Reply:
This function should probably just convert the param to a date if it exists, otherwise undefined. Then elsewhere you can use the date.toLocalString(), right?

In the other place you also did:
{new Date(expiresAtParam).toLocaleString()}

Which you can change to just `expiresAtParam.toLocaleString()`.

This function is still way too large and doing a lot of weird regex stuff which is not needed (we only support a number timestamp for the expiresAt param as per the README)